### PR TITLE
SRVKE-1446: Add eventing-istio-controller for channel-based mTLS

### PR DESCRIPTION
**Pull request by pierDipi**
- Add eventing-istio-controller
- scale down controller when feature flag is disabled and up when enabled

### CODE REVIEW

File: pkg/istio/eventingistio/istio.go [hunk 1]
```diff
@@ -132,3 +142,42 @@ func serviceMeshNetworkPolicies() []networkingv1.NetworkPolicy {
...
+	}
+	return scaleEventingIstioController(client, eventing, func(d *appsv1.Deployment) (bool, int32) {
+		return *d.Spec.Replicas > 0, 0
+	})
+}
+
+func scaleEventingIstioController(client client.Client, eventing *operatorv1beta1.KnativeEventing, shouldScaleFn func(d *appsv1.Deployment) (bool, int32)) error {
+	istioControllerName := types.NamespacedName{Namespace: eventing.GetNamespace(), Name: "eventing-istio-controller"}
+
+	overrides := eventing.GetSpec().GetWorkloadOverrides()
```
Comment (matzew) [inline pkg/istio/eventingistio/istio.go:160]: Just curious, we already have product documentation for the workload ovrrides?

But, good using that, instread of ha-replica

**Comment by pierDipi** [comment]
Unrelated failure

/test 4.12-upstream-e2e-mesh-aws-ocp-412

**Comment by pierDipi** [comment]
TestDestroyPodInflight

/test 4.12-upstream-e2e-mesh-aws-ocp-412

**Comment by pierDipi** [comment]
It needs https://github.com/openshift-knative/eventing/pull/239

**Comment by pierDipi** [comment]
The last run looks promising:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/2069/pull-ci-openshift-knative-serverless-operator-main-4.12-upstream-e2e-mesh-aws-ocp-412/1663532865570213888

There are too many tests in the mesh job that it times out

**Comment by pierDipi** [comment]
Even after https://github.com/openshift/release/pull/39844, the timeout is still 2h?

**Pull request merged**